src/main.ts

```
@@ -0,0 +1,165 @@
+import { app, BrowserWindow, dialog } from "electron";
+import { readFileSync } from "fs";
+import * as path from "path";
+import * as process from "process";
+
+enum OS {
+  WINDOW = process.platform === "darwin" ? 0 : 1,
+  MAC = process.platform !== "darwin" ? 0 : 1,
+}
+
+const permittedExt: Array<string> = [".tt", ".oa", ".oc"];
+const mainnetOpenCert = "https://www.opencerts.io/";
+const mainnetTradeTrust = "https://www.tradetrust.io/";
+
+let macFilePath: string | null = null;
+
+// Return filePath
+function getFilePath(): string | null {
+  if ((process.env.NODE_ENV || "").trim() === "development") {
+    return path.join(app.getAppPath(), "assets/demo.tt");
+  }
+
+  if (OS.MAC) {
+    return macFilePath;
+  }
+
+  if (OS.WINDOW) {
+    // process.argv[0] is exe path
+    // process.argv[1] is open file path
+    if (process.argv.length >= 2) {
+      const windowFilePath = process.argv[1];
+      return windowFilePath !== "." ? windowFilePath : null;
+    }
+  }
+  return null;
+}
+
+function validateFile(filePath: string): string | null {
+  const extIndex = permittedExt.indexOf(path.extname(filePath));
+  switch (extIndex) {
+    case 0:
+    case 1:
+      return mainnetTradeTrust;
+    case 2:
+      return mainnetOpenCert;
+    default:
+      return null;
+  }
+}
+
+function createWindow(verifier: string): BrowserWindow | null {
+  // Create the browser window.
+  let mainWindow: BrowserWindow;
+  if (BrowserWindow.getAllWindows().length > 0 && OS.MAC) {
+    mainWindow = BrowserWindow.getFocusedWindow();
+  } else {
+    mainWindow = new BrowserWindow({
+      width: 800,
+      height: 600,
+      titleBarStyle: "hidden",
+      autoHideMenuBar: true,
+      webPreferences: {
+        preload: path.join(__dirname, "preload.js"),
+      },
+    });
+  }
+
+  try {
+    mainWindow.loadURL(verifier);
+    return mainWindow;
+  } catch (e) {
+    return null;
+  }
```

there is an error so should return the error if possible, rather than return null

---

Added console.error(e) for developer to view more in-depth error message.
Return string exception, Programme will check if the instance is browserwindow / string before continuing